### PR TITLE
fix: parsing mention wrapped with styles inside link text

### PR DESCRIPTION
**Pull request by eh2077**
### Fixed Issues
$ https://github.com/Expensify/App/issues/19997

# Tests
1. Go to a chat and add a comment 
```
[_@username@expensify.com_](expensify.com)
```
2. Verify that the text `@username@expensify.com` of link is not rendered as mention and is italic 

# QA
Same as test

### CODE REVIEW

File: __tests__/ExpensiMark-HTML-test.js [hunk 1]
```diff
@@ -985,6 +985,26 @@ test('Test for @here mention without space or supported styling character', () =
     expect(parser.replace(testString)).toBe(resultString);
 });
 
+test('Test for mention inside link and wrapped by styles like italic, bold, strikethrough or code', () => {
```
Comment (s77rt) [inline __tests__/ExpensiMark-HTML-test.js:988]: Can you add tests for email as well e.g. `[_@username@expensify.com_](username@expensify.com)`

Also we have `Test for user mention inside link markdown` and `Test for user mention inside email markdown` tests, maybe we can merge those here.

**Review by s77rt — CHANGES_REQUESTED** [review]

**Review by s77rt — APPROVED** [review]
LGTM! :rocket: 

cc @luacmartins

**Review by luacmartins — APPROVED** [review]
LGTM

**Comment by johnmlee101** [comment]
Wasn't I suppose to be assigned?

**Comment by luacmartins** [comment]
I think so, not sure why Melvin assigned me instead 🤔

**Comment by s77rt** [comment]
I think auto assignment never worked correctly cross repos.

**Pull request merged**